lib/components/user/phone-change-form.tsx

```diff
@@ -0,0 +1,182 @@
+//   => invalid (red) alert.
+const phoneValidationSchema = yup.object({
+  phoneNumber: yup
+    .string()
+    .required()
+    .test(
+      'phone-number-format',
+      'invalidPhoneNumber', // not directly shown.
+      (value?: string) => value && isPossiblePhoneNumber(value)
+    )
+})
+
+interface Fields {
+  phoneNumber: string
+}
+
+export type PhoneChangeSubmitHandler = (
+  values: Fields | MouseEvent<Button>
+) => void
+
+interface Props {
+  isSubmitting: boolean
+  onCancel: () => void
+  onSubmit: PhoneChangeSubmitHandler
+  phoneFormatOptions: {
+    countryCode: string
+  }
+  showCancel?: boolean
+}
+
+type InnerProps = FormikProps<Fields> & Props
+
+const InnerPhoneChangeForm = ({
+  errors, // Formik
+  handleBlur, // Formik
+  handleChange, // Formik
+  isSubmitting,
+  onCancel,
+  phoneFormatOptions,
+  showCancel,
+  touched, // Formik
+  values // Formik
+}: InnerProps) => {
+  const intl = useIntl()
+  const formId = 'phone-change-form'
+  const showPhoneError = errors.phoneNumber && touched.phoneNumber
+
+  const handleEscapeKey = useCallback(
+    (e: KeyboardEvent<FormGroup>) => {
+      if (e.key === 'Escape' && showCancel && typeof onCancel === 'function') {
+        // Cancel editing when user presses ESC from the phone number field.
+        onCancel()
+        e.preventDefault()
+      }
+    },
+    [onCancel, showCancel]
+  )
+
+  const handlePhoneChange = useCallback(
+    (newNumber) =>
+      handleChange({
+        target: {
+          name: 'phoneNumber',
+          value: newNumber
+        }
+      }),
+    [handleChange]
+  )
+
+  return (
+    <FormGroup
+      // Handle ESC key from anywhere in this element.
+      onKeyDown={handleEscapeKey}
+      validationState={showPhoneError ? 'error' : null}
+    >
+      {/* Set up an empty Formik Form without inputs, and link inputs using the form id.
+          (A submit button within will incorrectly submit the entire page instead of just the subform.)
+          The containing Formik element will watch submission of the form. */}
+      <Form id={formId} noValidate />
+      <ControlLabel htmlFor="phone-number">
+        <FormattedMessage id="components.PhoneNumberEditor.prompt" />
+      </ControlLabel>
+      <ControlStrip>
+        <InlinePhoneInput
+          aria-invalid={showPhoneError}
+          aria-required
+          className="form-control"
+          country={phoneFormatOptions.countryCode}
+          form={formId}
+          id="phone-number"
+          name="phoneNumber"
+          onBlur={handleBlur}
+          onChange={handlePhoneChange}
+          placeholder={intl.formatMessage({
+            id: 'components.PhoneNumberEditor.placeholder'
+          })}
+          type="tel"
+          value={values.phoneNumber}
+        />
+
+        <Button
+          bsStyle="primary"
+          disabled={isSubmitting}
+          form={formId}
+          type="submit"
+        >
+          {isSubmitting ? (
```

Comment:
If someone enters an invalid validation code and clicks validate the button never goes back to regular!

Reply:
Good find! Fixed in https://github.com/opentripplanner/otp-react-redux/pull/799/commits/f8cb9b978a08b0ab7eab741df85bdfcfff037aa6.